contracts/VaultV3.vy

```diff
@@ -487,10 +493,16 @@ def _assess_share_of_unrealised_losses(strategy: address, assets_needed: uint256
 
 
 @internal
-def _redeem(sender: address, receiver: address, owner: address, shares_to_burn: uint256, strategies: DynArray[address, 10] = []) -> uint256:
+def _redeem(sender: address, receiver: address, owner: address, shares_to_burn: uint256, _strategies: DynArray[address, 10] = []) -> uint256:
     if sender != owner:
         self._spend_allowance(owner, sender, shares_to_burn)
 
+    strategies: DynArray[address, 10] = _strategies
+
+    queue_manager: address = self.queue_manager
+    if queue_manager != empty(address):
```

Comment:
This gets rid of the potential for anyone to pass a custom withdrawal queue in if there is a queue manager. Should it be if queue_manager != empty(address) and _strategies == []: 

That way someone can still customize the queue and since all strategies are permisioless anyway it shouldnt add extra risk

Reply:
It's a good question worth discussing. How much do we like people being able to choose custom queues. i think you are right and that the move to permissionless strategies means that security from queue ordering matters a lot less

Reply:
Also should there be a call to queue manager when a new strategy is added or revoked?

Reply:
I dont think it should call queue manager, the queue can only hold 10 strategies but the vault can hold as many as we want. The queue should be set by gov i would think. To line it up with the expected apr as well as how liquid a strategy is etc. for the easiest cheapest withdraws

Reply:
Sure, I suppose you could implement it in the adder. Addstrategy + and addToQueue on the queuemanager

Reply:
> This gets rid of the potential for anyone to pass a custom withdrawal queue in if there is a queue manager. Should it be if queue_manager != empty(address) and _strategies == []:
> 
> That way someone can still customize the queue and since all strategies are permisioless anyway it shouldnt add extra risk

What do you think to passing the strategy list in as a parameter to queue manager as well? That way the queue manager lets the user decide  if they can override the queue for max optionality

Reply:
> > This gets rid of the potential for anyone to pass a custom withdrawal queue in if there is a queue manager. Should it be if queue_manager != empty(address) and _strategies == []:
> > That way someone can still customize the queue and since all strategies are permisioless anyway it shouldnt add extra risk
> 
> What do you think to passing the strategy list in as a parameter to queue manager as well? That way the queue manager lets the user decide if they can override the queue for max optionality

Wouldnt that add alot of gas to be passing a 10 item array back and forth twice.

Vyper will pass the full array even if only one strategy is included so could add quite a bit of extra

Reply:
Yeh it would good point. What about a query to a shouldOverride view function in the queue manager? 

I am thinking that can be cheap because the queue manager doesnt need to use a storage slot to return the answer. it can just be a constant which would mean just a basic call

Reply:
So your thinking if the queue was empty it will just pull from the manager but if not it will check with the manager whether or not to allow the user to override? And if false then pull from the manager anyway

-> " the queue manager doesnt need to use a storage slot to return the answer."

Wouldnt this mean each vault would need its own queue_manager deployed specifically for it? 

I would imagine we would want to use the same queue manager for most of the vaults. Not a big issue to pull one bool from storage either way though

Reply:
Imagine a queue manager which always override. then the function would just return true regardless.

Reply:
https://github.com/flashfish0x/yearn-vaults-v3/pull/1/commits/59b85ca95d81f44e0c81e2cab827a2755a3895c2